test/src/test/java/org/corfudb/integration/LockIT.java

```diff
@@ -230,17 +235,25 @@ public void testMultipleLockClientsSameLock() throws Exception {
             }
 
             // Verify that only one client has acquired the lock
-            List<LockClient> clientsWithLock = getClientsThatAcquiredLock(lockId, clientIdToLockClient);
+            // We might need to verify for several cycles as
+            // the observable indicating the lock was acquired is
+            // triggered before the state update
+            List<LockClient> clientsWithLock;
+            do {
+                clientsWithLock = getClientsThatAcquiredLock(lockId, clientIdToLockClient);
+            } while (clientsWithLock == null || clientsWithLock.isEmpty());
+
             assertThat(clientsWithLock.size()).isEqualTo(1);
             LockClient clientWithLock = clientsWithLock.get(0);
 
-
             // Verify for 5 cycles that the lock is renewed
             for (int i=0; i < RENEW_CYCLES; i++) {
                 log.debug("***** Wait until lock is renewed");
                 // Wait for the renewal cycle + 1, and verify that the lock is still acquired by the same client
                 Sleep.sleepUninterruptibly(Duration.ofSeconds(LOCK_TIME_CONSTANT + 1));
-                clientsWithLock = getClientsThatAcquiredLock(lockId, clientIdToLockClient);
+                do {
```

Comment:
can we add the comment from line 238 here as well?

Reply:
Done.